### DISCUSSION TITLE
Mercury theme – Front page -> Filter by: Maintag (#each Maintags helper)

### DISCUSSION THREAD

**Discussion by candidexmedia** [GetPublii/Publii#1401] [Useful Resources] [answered]
I'm using this space to continue an [old forum discussion](https://forum.getpublii.com/topic/mercury-theme-front-page-filter-by-maintag-has-no-effect-on/) between much.fun, b0fh and I about the mercury theme filter.

It's also an extension of these old discussion posts:
- https://forum.getpublii.com/topic/mercury-maintag/
- https://forum.getpublii.com/topic/homepage-tag-list-not-contained-mercury-theme/

I wasn't able to add to our discussion because it was closed for replies, but good news @3muchfun : [a helper has been created to only display mainTags, and it's planned for v.0.45](https://github.com/GetPublii/Publii/issues/1329#issuecomment-1930943339).

Hopefully, the Mercury theme (and whichever other themes using the filter) will be updated in tandem to implement this new helper. If not, ping me, and I'll take a stab at updating it. I'm

**Comment by candidexmedia** [comment]
@3muchfun: I haven't updated my Publii app yet (long story), but you could test if the [new mainTag feature](https://github.com/GetPublii/Publii/issues/1329) works with the Mercury theme and latest Publii version? Doesn't look like the devs have updated the theme yet, but I wanted to see if the feature works.

To do so:

1. Go to your [sites location](https://getpublii.com/docs/publii-global-settings.html#fileslocation) > input > themes > mercury (or mercury-override), then open the `index.hbs` file in a text editor
2. Find the following code:
  ```handlebars
               {{#each tags}}
                  <li>
                     <button class="filter__item" data-filter=".{{slug}}">{{name}}</button>
                  </li>
               {{/each}}
  ```
3. Change the `tags` to `mainTags` as such:
  ```handlebars
               {{#each mainTags}}
                  <li>
                     <button class="filter__item" data-filter=".{{slug}}">{{name}}</button>
                  </li>
               {{/each}}
  ```

3. Save the file, then preview your homepage via the Publii GUI. 

4. If it's working properly, only your mainTags should be displayed.

**Comment by bobmitro** [comment, marked as answer]
Thanks, @candidexmedia, for your assistance.
FYI: I updated the Mercury theme to version [2.7.0.0](https://marketplace.getpublii.com/themes/mercury/), and it should run smoothly now :)

**Reply by candidexmedia** [reply]
Sweet, thank you!

**Comment by 3muchfun** [comment]
Hi there,
thanks for updating the theme, but after updating Publii 45.2 with Mercury version 2.7.0.0, the filter on the homepage is completely empty. Each of my posts has one tag which is defined as maintag (I thought).
Any ideas what went wrong, @candidexmedia, @bobmitro ?

**Reply by bobmitro** [reply]
Have you made any custom modifications to your theme by overwriting the original theme files in Publii theme?

**Reply by 3muchfun** [reply]
No, I haven't. I put some CSS in the Custom CSS field, only.

**Reply by bobmitro** [reply]
I have no idea why it doesn't work for you. Please send me (via email) a copy of your site.

**Reply by 3muchfun** [reply]
OK, Bob, I will do it later on.

**Reply by 3muchfun** [reply]
Hi, @bobmitro, which files do you need? the input folder as a .zip file?
